app/routes.py

```diff
@@ -1,2 +1,218 @@
+        "task": id.to_json()
+    }
+
+    path = "https://slack.com/api/chat.postMessage"
+    text_response = f"Someone just completed the task {id.title}"
+    query_params = {
+        "channel": "task-notifications",
+        "text": text_response
+    }
+    header = {
+        "Authorization": f"Bearer {os.environ.get('SLACK_TOKEN')}"
+    }
+    response = requests.post(path, params=query_params, headers=header)
+    
+
+    return jsonify(response_body), 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def make_task_incomplete(task_id):
+    id = Task.query.get(task_id)
+
+    if id is None:
+        return make_response("", 404)
+
+    id.completed_at = None
+    db.session.commit()
+    response_body = {
+        "task": id.to_json()
+    }
+    return jsonify(response_body), 200
+    
+#Goals routes***********************************************************************************************************************
+@goals_bp.route("", methods=["GET", "POST"])
+def handle_goals():
+    if request.method == "POST":
+        req_body = request.get_json()
+        if "title" not in req_body:
+            return {
+                "details": "Invalid data"
+            }, 400
+        new_goal = Goal(title = req_body["title"])
+        db.session.add(new_goal)
+        db.session.commit()
+        response_body = {
+            "goal": new_goal.goal_json()
+        }
+        return jsonify(response_body), 201
+    
+    if request.method == "GET":
+        all_goals = Goal.query.all()
+        response_body = []
+        for any_goal in all_goals:
+            response_body.append(any_goal.goal_json())
+        return jsonify(response_body), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET", "DELETE", "PUT"], strict_slashes=False)
+def goal_id_functions(goal_id):
+    a_goal = Goal.query.get(goal_id)
+
+    if a_goal is None:
+        return make_response("", 404)
+    
+    if request.method == "GET":
+        return {
+            "goal": a_goal.goal_json()
+        }, 200
+    
+    elif request.method == "PUT":
+        info = request.get_json()
+        a_goal.title = info["title"]
+        db.session.commit()
+        return {
+            "goal": a_goal.goal_json()
+        }, 200
+    
+    elif request.method == "DELETE":
+        db.session.delete(a_goal)
+        db.session.commit()
+        return {
+            "details": f"Goal {a_goal.goal_id} \"{a_goal.title}\" successfully deleted"
+        }, 200
+
+#WAVE 6*********************************************************************************************************************
+
+@goals_bp.route("/<goals_id>/tasks", methods=["POST"], strict_slashes=False)
+def tasks_within_goals(goals_id):
+    goals_id = int(goals_id)
+    request_body = request.get_json()
+    multiple_task_ids = request_body["task_ids"]
+    
+    if goals_id is None:
+        return make_response("", 404)
+
+    list_of_ids = []
+    for ids in multiple_task_ids:
```

Comment:
This is a nitpick but I suggest using singular names for most variables that represent a single value. 
```suggestion
    for id in multiple_task_ids:
```